streams/src/components/mp4-muxer/boxbuilder.ts

```diff
@@ -150,48 +173,16 @@ export class BoxBuilder {
     // a track in the MP4 file. For each track, a 'trak' box is added to the
     // 'moov' box and a 'trex' box is added to the 'mvex' box.
 
-    this.trackIdMap = {}
     this.sequenceNumber = 0
-    this.ntpPresentationTime = 0
-
-    let trackId = 0
-    this.trackData = []
-
-    sdp.media.forEach((media) => {
-      if (media.rtpmap === undefined) {
-        return
-      }
-
-      const payloadType = media.rtpmap.payloadType
-      const encoding = media.rtpmap.encodingName
-
-      if (formatDefaults[encoding] !== undefined) {
-        // We know how to handle this encoding, add a new track for it, and
-        // register the track for this payloadType.
-        this.trackIdMap[payloadType] = ++trackId
-
-        // Mark the video track
-        if (media.type.toLowerCase() === 'video') {
-          this.videoTrackId = trackId
-        }
+    this.videoStartTime = 0
 
-        // Extract the settings from the SDP media information based on
-        // the encoding name (H264, MPEG4-GENERIC, ...).
-        const settings = formatDefaults[encoding](media, date, trackId)
-        media.mime = settings.mime // add MIME type to the SDP media
-        media.codec = settings.codec // add human readable codec string to the SDP media
+    this.trackData = {}
 
-        const trackData = createTrackData()
-        trackData.clockrate = media.rtpmap.clockrate
-        // Set default frame duration (in ticks) for later use
-        trackData.defaultFrameDuration = settings.defaultFrameDuration
-
-        this.trackData.push(trackData)
-
-        const trak = this.trak(settings)
-        moov.append(trak)
-        mvex.append(new Box('trex', { track_ID: trackId }))
-      }
+    tracks.forEach((track) => {
+      this.trackData[track.payloadType] = createTrackData(track)
+      const trak = this.trak(track)
```

Comment:
Intentional misspelling?

Reply:
The box is called "trak" (the identifier)